extension/json/include/json_scan.hpp

```diff
@@ -296,6 +296,17 @@ struct JSONScan {
 		return lstate.GetBatchIndex();
 	}
 
+	static unique_ptr<NodeStatistics> JSONScanCardinality(ClientContext &context, const FunctionData *bind_data) {
+		auto &data = (JSONScanData &)*bind_data;
+		idx_t per_file_cardinality;
+		if (data.stored_readers.empty()) {
+			per_file_cardinality = 42;
```

Comment:
Are these real magic numbers, or just here for the joke?

Reply:
The CSV reader has a witty comment if it has to guess the number of rows in a CSV file without knowing the file size, which also defaults to 42. As for the the other estimate, I've estimated the average size of a JSON object to be 420 bytes mostly because it's 42 * 10, but also because the size of a JSON row in the `lineitem` table is pretty similar at an average of ~360 bytes.

This is obviously not a great estimate, but it's better than estimating nothing because that will default to an estimated cardinality of 0.

Reply:
Maybe we can do the number of columns * 42 instead? Or are the number of columns not known yet at this stage?

Reply:
They are known at this time if this is `read_(nd)json(_auto)`, but in JSON you can have one top-level column which is deeply nested, so there's really no guarantees to the size of column. Users may also specify that a deeply nested column should not be unnested by specifying it to be type `JSON`, which means that this column is huge. I can try to make the heuristic slightly smarter but it will be off by a lot in many cases. The best way to do this is to measure the size of the first JSON, which we could do when auto-detecting, but not when columns are manually specified.